pyproject.toml

```diff
@@ -1,3 +1,75 @@
+[project]
+dynamic = ["version", "entry-points", "scripts"]
+name = "cms"
+license = "AGPL-3.0-or-later"
+maintainers = [ { name = "The CMS development team", email = "contestms@googlegroups.com" } ]
+readme = "README.md"
+description = "A contest management system and grader for IOI-like programming competitions"
+requires-python = ">=3.11"
+keywords = ["ioi", "programming contest", "grader", "management system"]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Natural Language :: English",
+    "Operating System :: POSIX :: Linux",
+    "Programming Language :: Python :: 3.11",
+]
+
+dependencies = [
+    "tornado==4.5.3",         # http://www.tornadoweb.org/en/stable/releases.html
+    "psycopg2==2.9.10",       # http://initd.org/psycopg/articles/tag/release/
+    "sqlalchemy>=1.3,<1.4",   # http://docs.sqlalchemy.org/en/latest/changelog/index.html
```

Comment:
I'm thinking we could pin all dependencies to an exact version, using `==` everywhere. What do you think?

Reply:
I don't think we should. The dependencies in `pyproject.toml` should be bounds on what is believed to work, while `constraints.txt` should tell what we really tested and want to install CMS with.

Also, I am not sure how useful is the stream of PRs churned out by dependabot.